R/aggregate.R

```diff
@@ -0,0 +1,166 @@
+##' This function calculates the robust summarisation for each feature
+##' (protein). Note that the function assumes that the intensities in
+##' input `e` are already log-transformed.
+##'
+##' @title Return the Robust Expression Summary of a matrix
+##' @param x A feature by sample `matrix` containing quantitative
+##'     data with mandatory `colnames` and `rownames`.
+##' @param ... Additional arguments passed to [MASS::rlm()].
+##' @return `numeric()` vector of length `ncol(x)` with robust
+##'     summarised values.
+##' 
+##' @author Adriaan Sticker, Sebastian Gibb and Laurent Gatto
+##' 
+##' @export
+##' 
+##' @importFrom MASS rlm
+##' 
+##' @examples
+##' x <- matrix(rnorm(30), nrow = 3)
+##' colnames(x) <- letters[1:10]
+##' rownames(x) <- LETTERS[1:3]
+##' robustSummary(x)
+robustSummary <- function(x, ...) {
+    stopifnot(!is.null(colnames(x)))
```

Comment:
Actually `stopifnot` is a wrapper around a lot of code and will throw a more or less useless error message to the user. Would it be better to add `if (!is.null(colnames(x)) stop("colnames must not be empty")` here and the row below?

Reply:
Done